### DISCUSSION TITLE
Is there an API way to get a list of all open windows/workspaces?

### DISCUSSION THREAD

**Discussion by sketchbuch** [microsoft/vscode-discussions#1815] [Extension Development QnA] [answered]
For my extension, [Workspace Sidebar](https://github.com/sketchbuch/vsc-workspace-sidebar), I would like to be able to mark open workspaces visually so someone can see which ones are open in other windows. I couldn't see any way in the API to get access to this kind of information.

**Comment by alefragnani** [comment, marked as answer]
Hi @sketchbuch ,

AFAIK, there is no API available. 

I had a similar idea for my Project Manager extension, and back then I thought about controlling it myself inside the extension, using a shared memento in the global storage. In the end, the idea lost priority and I didn’t code anything.

Hope this helps

**Comment by sketchbuch** [comment]
I was just looking, but will give up on this idea. I can tell when a workspace opens but there is no way to know if a workspace closes or if a window closes. I found some requests for listening for window close but or for listen to a command (which might also work) but both were rejected.
